### DISCUSSION TITLE
[Feature]: Callable for register

### DISCUSSION THREAD

**Discussion by db0** [django-wiki/django-wiki#1309] [Ideas]
### I've already checked other feature proposals

Yes

### Is your feature request related to a problem?

Yes

### Describe your problem.

I want to be able to reject new user registrations based on bespoke rules

### Describe your feature request.

I would like to be able to call a function on a new user registration, with the arguments of the username and email used. I want to then be able to reject the registration if I return false

### Anything else we should know.

If you tell me where the registration functions are, I can look into sending a PR.

**Comment by oscarmcm** [comment]
@db0 is this solved? (from this https://github.com/django-wiki/django-wiki/discussions/1308#discussioncomment-7152144)

**Reply by db0** [reply]
No that is irrelevant. This is a new feature. I want to be able to call a custom function when a new user registers and possibly block the registration.

**Reply by oscarmcm** [reply]
Have you checked https://docs.djangoproject.com/en/4.2/topics/signals/ ?


While `signals` are not widely recommended, using this for prevent user registration will not hurt anyone.

https://simpleisbetterthancomplex.com/tutorial/2016/07/28/how-to-create-django-signals.html

**Reply by db0** [reply]
Well, I'm guessing this will be the approach the wiki callable might take, but I don't think I can do this myself without the callable existing already.
